main.tf

```diff
@@ -103,3 +104,29 @@ module "cbr_rule" {
     }]
   }]
 }
+
+##############################################################################
+# Event Notification
+##############################################################################
+
+# Create IAM Authorization Policies to allow SM to access event notification
+resource "ibm_iam_authorization_policy" "en_policy" {
+  count                       = var.enable_event_notification == false || var.skip_en_iam_authorization_policy ? 0 : 1
+  source_service_name         = "secrets-manager"
+  source_resource_group_id    = var.resource_group_id
+  target_service_name         = "event-notifications"
+  target_resource_instance_id = regex(".*:(.*)::", var.existing_en_instance_crn)[0]
+  roles                       = ["Event Source Manager"]
+  description                 = "Allow all Secrets Manager instances in the resource group ${var.resource_group_id} to access the event-notification instance GUID ${regex(".*:(.*)::", var.existing_en_instance_crn)[0]}"
```

Comment:
"Allow all Secrets Manager instances in the resource group ${var.resource_group_id} 'Event Source Manager' role access on the Event Notification instance GUID ${regex(".*:(.*)::", var.existing_en_instance_crn)[0]}"